sql/src/main/java/io/crate/metadata/cluster/AlterTableClusterStateExecutor.java

```diff
@@ -0,0 +1,437 @@
+                    supportedSettings.add(IndexMetaData.SETTING_AUTO_EXPAND_REPLICAS);
+                    TableParameter parameterWithFilteredSettings = new TableParameter(request.settings(), supportedSettings);
+
+                    currentState = updateSettings(currentState, parameterWithFilteredSettings.settings(), concreteIndices);
+                    currentState = updateMapping(currentState, request, concreteIndices);
+                }
+            }
+        } else {
+            currentState = updateMapping(currentState, request, concreteIndices);
+            currentState = updateSettings(currentState, request.settings(), concreteIndices);
+        }
+
+        return currentState;
+    }
+
+
+    private ClusterState updateMapping(ClusterState currentState, AlterTableRequest request, Index[] concreteIndices) throws Exception {
+        if (request.source() == null) {
+            return currentState;
+        }
+        Map<Index, MapperService> indexMapperServices = new HashMap<>();
+        for (Index index : concreteIndices) {
+            final IndexMetaData indexMetaData = currentState.metaData().getIndexSafe(index);
+            if (indexMapperServices.containsKey(indexMetaData.getIndex()) == false) {
+                MapperService mapperService = indicesService.createIndexMapperService(indexMetaData);
+                indexMapperServices.put(index, mapperService);
+                // add mappings for all types, we need them for cross-type validation
+                mapperService.merge(indexMetaData, MapperService.MergeReason.MAPPING_RECOVERY, false);
+            }
+        }
+
+        PutMappingClusterStateUpdateRequest updateRequest = new PutMappingClusterStateUpdateRequest()
+            .ackTimeout(request.timeout()).masterNodeTimeout(request.masterNodeTimeout())
+            .indices(concreteIndices).type(Constants.DEFAULT_MAPPING_TYPE)
+            .source(request.source());
+
+        return metaDataMappingService.putMappingExecutor.applyRequest(currentState, updateRequest, indexMapperServices);
+    }
+
+    /**
+     * The logic is taken over from {@link MetaDataUpdateSettingsService#updateSettings(UpdateSettingsClusterStateUpdateRequest, ActionListener)}
+     */
+    private ClusterState updateSettings(final ClusterState currentState, final Settings settings, Index[] concreteIndices) {
+
+        final Settings normalizedSettings = Settings.builder().put(settings).normalizePrefix(IndexMetaData.INDEX_SETTING_PREFIX).build();
+        Settings.Builder settingsForClosedIndices = Settings.builder();
+        Settings.Builder settingsForOpenIndices = Settings.builder();
+        final Set<String> skippedSettings = new HashSet<>();
+
+        //TODO Do we need all the modifications of the settings ?
+        indexScopedSettings.validate(
+            normalizedSettings.filter(s -> Regex.isSimpleMatchPattern(s) == false), // don't validateSettings wildcards
```

Comment:
I think we don't have to filter anymore at this point, all should be fine already